docs/general.md

```diff
@@ -0,0 +1,133 @@
+# General Debugr architecture
+
+The Debugr core is very simple and only does a couple of things.
+Most of the actual work is delegated to various plugins. This document
+describes the core concepts of how the Debugr core works.
+
+(todo: explain log entries, plugin types, log levels)
+
+## Log Entries
+
+Log entries are objects which represent data to be logged, along with some
+additional metadata. Log entries can be _generic_, meaning the data has no
+predefined semantics, or they can be _specialised_ to represent a specific
+type of event or information. Whenever you call `logger.log()` or any of the
+shortcuts a new log entry is created. Some plugins will produce entries
+from e.g. a library they integrate with, and other plugins will consume
+entries in order to produce some kind of output.
+
+The shape of a log entry looks like this:
+
+```typescript
+export type LogEntry = {
+  type?: EntryType;
+  level: LogLevel;
+  globalContext: TGlobalContext;
+  taskContext?: TTaskContext;
+  message?: string;
+  data?: Record<string, any>;
+  error?: Error;
+  ts: Date;
+};
+```
+
+The `Logger` class has an `add()` method, which expects an almost complete
+log entry object - it will only supply the global and per-task context and
+the timestamp. This method is typically used by plugins to log a _specialised_
+log entry. The `log()` method (and its shortcuts) constructs a _generic_ log entry
+from the supplied arguments and then delegates to `add()`. Notably, the `log()`
+method will apply `printf`-style formatting to the log message, if a `[format, ...args]`
+tuple is specified as the message. No other processing on the log entries is done
+in either method - everything else happens in plugins.
+
+## Log levels
+
+The `LogLevel` type is a `const enum`, meaning that in the compiled code
```

Comment:
nerikal jsi, ze je tu nejake uskali? + mozna vysvetlit jake log levely jsou defaultni

Reply:
jo, pravda

Reply:
A taky by chtelo nekde rozepsat k default log levelum i informaci k cemu je jaky level zamyslen

Reply:
nechtel bys tohle zkusit nastrelit ty? klidne mi to jen napis na slack, ale ja treba nevim uplne jakej vnimas rozdil mezi `debug` a `trace` nebo `error` a `fatal`, ty v tom evidentne nejakou logiku mas, tak bych uplne nevymejslel kolo..